Public/Src/Cache/ContentStore/Library/Sessions/ContentSessionBase.cs

```diff
@@ -159,15 +170,15 @@ protected ContentSessionBase(string name)
             CancellationToken token,
             UrgencyHint urgencyHint = UrgencyHint.Nominal)
         {
-            // TODO: add support for PerformOperation with Task<IEnumerable<Task<Indexed<ResultBase>>>>
-
-            using (var cancellableContext = TrackShutdown(context, token))
-            {
-                using (_counters[ContentSessionBaseCounters.PlaceFileBulk].Start())
-                {
-                    return await PlaceFileCoreAsync(cancellableContext, hashesWithPaths, accessMode, replacementMode, realizationMode, urgencyHint, _counters[ContentSessionBaseCounters.PlaceFileBulkRetries]);
-                }
-            }
+            return WithOperationContext(
+                context,
+                token,
+                operationContext => operationContext.PerformNonResultOperationAsync(
+                    Tracer,
+                    () => PlaceFileCoreAsync(operationContext, hashesWithPaths, accessMode, replacementMode, realizationMode, urgencyHint, _counters[ContentSessionBaseCounters.PlaceFileBulkRetries]),

```

Comment:
>PlaceFileCoreAsync [](start = 26, length = 18)

Side-Note: Some concern about debuggability and callstack bloat with all the levels of lambda indirections, but that applies to the whole cache code..